### PR TITLE
[Servers] Add liveness and readiness probes

### PR DESCRIPTION
**Pull request by Bobgy**
/assign @IronPan @rmgogogo 
Part of https://github.com/kubeflow/pipelines/issues/3756

Added probes for UI, api, visualization servers. Operators will be added in a separate PR.
I verified this workaround works properly when istio mutual TLS is enabled.

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -38,3 +38,7 @@ EXPOSE 3000
 RUN npm run build
 ENV API_SERVER_ADDRESS http://localhost:3001
 CMD node dist/server.js ../client/ 3000
+
+# HACK: curl is required for running liveness/readiness probe inside the container
```
Comment (Ark-kun) [inline frontend/Dockerfile:42]: JFYU: alpine has `wget`

Comment (Bobgy) [reply frontend/Dockerfile:42]: Ohh, thanks! I will switch to wget then.

**Comment by rmgogogo** [comment]
/lgtm
/approve

thanks

**Pull request merged**